controllers/argocd/configmap.go

```diff
@@ -628,9 +628,39 @@ func (r *ReconcileArgoCD) reconcileRedisConfiguration(cr *argoprojv1a1.ArgoCD) e
 	if err := r.reconcileRedisHAConfigMap(cr); err != nil {
 		return err
 	}
+	if err := r.reconcileRedisHAHealthConfigMap(cr); err != nil {
+		return err
+	}
 	return nil
 }
 
+// reconcileRedisHAConfigMap will ensure that the Redis HA Health ConfigMap is present for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileRedisHAHealthConfigMap(cr *argoprojv1a1.ArgoCD) error {
+	cm := newConfigMapWithName(common.ArgoCDRedisHAHealthConfigMapName, cr)
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, cm.Name, cm) {
+		if !cr.Spec.HA.Enabled {
+			// ConfigMap exists but HA enabled flag has been set to false, delete the ConfigMap
+			return r.Client.Delete(context.TODO(), cm)
+		}
+		return nil // ConfigMap found with nothing changed, move along...
+	}
+
+	if !cr.Spec.HA.Enabled {
```

Comment:
I would check for enabled first and exit early if ha is not enabled.

Reply:
The existing code (from before this PR) will delete the config map if it exists and HA is not enabled.  I think we want to keep that behaviour.